Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -11,6 +11,9 @@ import BraintreeCore
 
     /// A nonce to be verified by ThreeDSecure
     public var nonce: String?
+    
+    /// Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+    public var customFields: [String: Any]? = [:]
```

Comment:
Similar comment [to here](https://github.com/braintree/braintree_android/pull/1044/files#r1650990591), with making this a more standard type that merchants use to model JSON - `Encodable`. A dictionary will also work though.

Which do you think is better merchant dev-ex?

Reply:
I just got more information that we'll be supporting no nested JSON values and n number of custom values. I'm waiting on clarification to determine if we're expecting only String values.

Reply:
Confirmed we can enforce string key values. I think [String: String] is fine. The merchant will be setting these values in their control panel. Btw, the Cardinal team is taking anything but stringifying everything coming through the gateway.